decidim-core/app/helpers/concerns/decidim/user_role_checker.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Decidim
+  module UserRoleChecker
+    # Shared behaviour for signed_in admins
+    extend ActiveSupport::Concern
+
+    private
+
+    def user_has_any_role?(user, participatory_space = nil)
+      return false unless user
+      return true if user.admin
+      return true if user.roles.any?
+      return true if participatory_process_user_role?(user, participatory_space)
+      return true if assembly_user_role?(user, participatory_space)
+      return true if conference_user_role?(user, participatory_space)
+
+      false
+    end
```

Comment:
just a matter of style. I would say this version would reduce the number of exit points from this method to 2, instead of 7.

```suggestion
    def user_has_any_role?(user, participatory_space = nil)
      return false unless user

      [
        user.admin,
        user.roles.any?,
        participatory_process_user_role?(user, participatory_space),
        assembly_user_role?(user, participatory_space),
        conference_user_role?(user, participatory_space)
      ].any?
    end
```

Reply:
This is not fixed